File: projects/04/factorial.rb
#! /usr/bin/ruby
# factorial.rb computes the factorial of a given number
#
# Begun by: Dr. Nelesen, for CS 214 at Calvin College.
# Completed by: Zach Wibbenmeyer
# Date: March 7, 2016
############################################################

# computeFactorial() method - computes the factorial of an integer
# @param: n
# @return: a float
# Precondition: n must be an integer
def computeFactorial(n)
	# Create a local variable
	answer = 1.0

	# Main for loop for calculating factorial
	for i in 2 .. n
		answer *= i

	end
	return answer
end

# Main program
if __FILE__ == $0

	# Prompt user for a number
	print "To compute n!, enter n: "
	# Receive number
	num = gets.chomp.to_f
	# Compute factorial using method
	puts computeFactorial(num)
end

File: projects/04/Factorial.java
/* Factorial.java calculates the factorial of a given double
 *
 * Input: n, a double
 * Begun by: Nelesen, for CS 214 at Calvin College.
 * Completed by: Zach Wibbenmeyer
 * Date: March 7, 2016
 ********************************************************/

import java.util.Scanner;		// Get input

/* Main class for computing the factorial of a double */
public class Factorial {

	/* factorial() method - computes the factorial of a double
	 * @param: n (type = double);
	 * @return: a double
	 * Precondition: n must be an integer double
	 */
	public static double factorial(double n) {
		// Create a local variable
		double answer = 1.0;

		// Main for loop for factorial computation
		for (int i = 2; i <= n; i++) {
			answer *= i;
		}
		return answer;
	}

	// Main program
	public static void main(String[] args) {
		// Link to the keyboard
		Scanner keyboard = new Scanner(System.in);
		// Prompt user for a number
		System.out.println("To compute n!,");
		System.out.println("enter n: ");

		// Compute the factorial using the method above
		double num = keyboard.nextDouble();
		System.out.println(factorial(num));
	}
}